service/controller/app/v1/resource/chart/current.go

```diff
@@ -50,6 +55,12 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	} else if err != nil {
 		return nil, microerror.Mask(err)
 	}
+	if errors.Is(ctx.Err(), context.DeadlineExceeded) {
```

Comment:
Does it really work? doesn't we have to wrap it with `select` statement so it would return timeout errors by a channel?
Example in golang docs; 
```go
package main

import (
	"context"
	"fmt"
	"time"
)

func main() {
	// Pass a context with a timeout to tell a blocking function that it
	// should abandon its work after the timeout elapses.
	ctx, cancel := context.WithTimeout(context.Background(), 50*time.Millisecond)
	defer cancel()

	select {
	case <- time.After(1 * time.Second):
		fmt.Println("overslept")
	case <-ctx.Done():
		fmt.Println(ctx.Err()) // prints "context deadline exceeded"
	}

}
```

I did a similar thing as this code suggests here. https://play.golang.org/p/A9X7iYHCqTW
But it doesn't work.

Reply:
It worked when I tested the migration resources in cluster-operator. But I'll do some more testing tomorrow.

Reply:
@tomahawk28 Your example works for me.

This was tested in cluster-operator but doing another test to be sure. I'd prefer to keep the code as it is because its simpler.

Reply:
The question to me is, it would fall into context timeout error if it takes more than 30 seconds. 
The current structure suggests it would only know that context timeout happened after 30 seconds. 
Can we maybe test that situation?

Reply:
The context timeout is 3 seconds not 30 seconds. We need to use a short timeout because we need to make a lot of API calls to fetch the current state.

I'm going to test it in ginger with a working TC and a broken TC. I think that's the best way to simulate the current problems.

Reply:
Sorry, my knowledge around here maybe not enough. 
What I mean above, should we expect timeout (3 seconds) to happen with the `select` statement? 
```
	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
	defer cancel()
        c := make(chan int)
        go func() { 
        // Working with tenant cluster       
  //...
       // Done     
       <- interface{}
       } () 

	select {
	case <- c 
                // Done normally, let's move on. 
	case <-ctx.Done():
		// timeout happened first, canceled the reconciliation. 
	}
```

Reply:
@tomahawk28 Thanks exactly right. <3 This needed the channel and signal handling. 🤦‍♂ 

PTAL Tested on `ginger`.